designs/Net5PlatformAndPlatformVersion.md

```diff
@@ -0,0 +1,177 @@
+
+Source: src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs
+
+### Fields
+
+Three additional fields will be added:
+
+```csharp
+        /// <summary>
+        /// Framework Platform (net5.0+)
+        /// </summary>
+        public string Platform
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Framework Platform Version (net5.0+)
+        /// </summary>
+        public Version PlatformVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// True if platform is non-empty (net5.0+)
+        /// </summary>
+        public bool HasPlatform
+        {
+            get { return !string.IsNullOrEmpty(Platform); }
+        }
+
+        /// <summary>
+        /// True if platform version is non-empty (net5.0+)
+        /// </summary>
+        public bool HasPlatformVersion
+        {
+            get { return IsNet5Era && !(PlatformVersion.Major == 0 && PlatformVersion.Minor == 0); }
+        }
+```
+
+### Constructors
+
+An additional constructor should be added that accepts `Version platformVersion` as its 4th argument. The current code in `NuGetFramework/3` should be moved to this new constructor, and `string profile` should be renamed to `string profileOrPlatform`. This field will be EITHER a legacy profile or, if net5.0+, a platform identifier.
+
+```csharp
+        public NuGetFramework(string framework, Version version, string profileOrPlatform)
+            : this(framework, version, profileOrPlatform, FrameworkConstants.EmptyVersion)
+        {
+        }
+
+        private const int Version5 = 5;
+
+        public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, string profileOrPlatform, Version platformVersion)
+        {
+            if (frameworkIdentifier == null)
+            {
+                throw new ArgumentNullException("frameworkIdentifier");
+            }
+
+            if (frameworkVersion == null)
+            {
+                throw new ArgumentNullException("frameworkVersion");
+            }
+
+            _frameworkIdentifier = frameworkIdentifier;
+            _frameworkVersion = NormalizeVersion(frameworkVersion);
+
+            IsNet4Era = (_frameworkVersion.Major >= Version5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, _frameworkIdentifier));
+
+            _frameworkProfile = IsNet5Era ? profileOrPlatform ?? string.Empty : string.Empty;
+            Platform = IsNet5Era ? profileOrPlatform ?? string.Empty : string.Empty;
+            PlatformVersion = NormalizeVersion(platformVersion ?? FrameworkConstants.EmptyVersion);
+        }
+```
+
+### GetDotNetFrameworkName
+
+This method needs to be updated to append the Platform and PlatformVersion. I believe the format is supposed to be:
+
+```csharp
+string.Format(CultureInfo.InvariantCulture, "Platform={0},PlatformVersion={0}", framework.Platform, framework.PlatformVersion);

```

Comment:
Maybe we need to check with the .NET team. I'm not sure `Platform=` is right. Might be `PlatformIdentifier=`.

Reply:
`TargetProfileIdentifier` uses `Profile=`, which is why I did it this way.